frontend/src/main/scala/bloop/data/Project.scala

```diff
@@ -227,27 +227,29 @@ object Project {
       logger: Logger
   ): Project = {
     def enableSemanticDB(options: List[String], pluginPath: AbsolutePath): List[String] = {
+      val workspaceDir = project.workspaceDirectory.getOrElse(configDir.getParent)
+      val baseSemanticdbOptions = List(
+        "-P:semanticdb:failures:warning",
+        "-P:semanticdb:synthetics:on",
+        "-Xplugin-require:semanticdb"
+      )
+      // TODO: Handle user-configured `targetroot`s inside Bloop's compilation
+      // engine so that semanticdb files are replicated in those directories
       val hasSemanticDB = hasSemanticDBEnabledInCompilerOptions(options)
-      if (hasSemanticDB) options
-      else {
-        val workspaceDir = project.workspaceDirectory.getOrElse(configDir.getParent)
-        // TODO: Handle user-configured `targetroot`s inside Bloop's compilation
-        // engine so that semanticdb files are replicated in those directories
-        val semanticdbScalacOptions = List(
-          "-P:semanticdb:failures:warning",
-          s"-P:semanticdb:sourceroot:$workspaceDir",
-          "-P:semanticdb:synthetics:on",
-          "-Xplugin-require:semanticdb",
-          s"-Xplugin:$pluginPath"
-        )
-
-        (options ++ semanticdbScalacOptions.toList).distinct
-      }
+      val pluginOption = if (hasSemanticDB) Nil else List(s"-Xplugin:$pluginPath")
+      val sourceRootOption =
+        if (hasSemanticdbSourceRoot(options)) {
```

Comment:
I would filter out existing sourceroot settings if there are any. It's critical that the sourceroot matches the same directory as `.bloop`, otherwise code navigation wouldn't work.

Reply:
Sure can do. @jvican what do you think?

Reply:
Ok, change to replacing sourceroot.